core/src/main/scala/sttp/client3/testing/SttpBackendStub.scala

```diff
@@ -118,8 +118,27 @@ class SttpBackendStub[F[_], +P](
       */
     def thenRespondCyclicResponses[T](responses: Response[T]*): SttpBackendStub[F, P] = {
       val iterator = Iterator.continually(responses).flatten
-      thenRespond(iterator.next)
+      thenRespond(iterator.next())
     }
+
+    /** Thread-safe version of `thenRespondCyclic`
+      */
+    def thenRespondCyclicAtomically[T](body1: T, bodies: T*): SttpBackendStub[F, P] = {
```

Comment:
looks good :) maybe we can just improve the existing `thenRespondCyclic` functions? Once we have thread-safe versions, I don't think there's a reason to keep them around :)

Reply:
Yup, will replace the existing ones instead.

Reply:
Updated. The old impl is now replaced by the atomic impl. Now I need to deal with failed tests. There are so many tests to run and I've been getting OOM towards the end. Need to figure that out too.